--- faac-1.28/common/mp4v2/mpeg4ip.h.orig	2009-01-26 23:42:35.000000000 +0100
+++ faac-1.28/common/mp4v2/mpeg4ip.h	2010-03-10 23:42:21.000000000 +0100
@@ -120,14 +120,6 @@
 #endif
 #include <sys/param.h>
 
-#ifdef __cplusplus
-extern "C" {
-#endif
-char *strcasestr(const char *haystack, const char *needle);
-#ifdef __cplusplus
-}
-#endif
-
 #define OPEN_RDWR O_RDWR
 #define OPEN_CREAT O_CREAT 
 #define OPEN_RDONLY O_RDONLY
