--- goom2k4-0/configure.orig	2005-02-07 14:49:28.000000000 +0100
+++ goom2k4-0/configure	2010-11-28 22:50:21.956572524 +0100
@@ -21082,17 +21082,17 @@
 esac
 
 case "$host" in
-i*86-*-*)
-	cat >>confdefs.h <<\_ACEOF
-#define HAVE_MMX 1
-_ACEOF
-
-	cat >>confdefs.h <<\_ACEOF
-#define CPU_X86 1
-_ACEOF
-
-	HAVE_MMX="yes"
-	;;
+#i*86-*-*)
+#	cat >>confdefs.h <<\_ACEOF
+##define HAVE_MMX 1
+#_ACEOF
+#
+#	cat >>confdefs.h <<\_ACEOF
+##define CPU_X86 1
+#_ACEOF
+#
+#	HAVE_MMX="yes"
+#	;;
 
 powerpc-*-*)
   	CCASFLAGS=-force_cpusubtype_ALL
