diff -uarN libdc1394-1.2.2.orig/Makefile.in libdc1394-1.2.2/Makefile.in
--- libdc1394-1.2.2.orig/Makefile.in	2007-08-03 10:09:07.000000000 +0200
+++ libdc1394-1.2.2/Makefile.in	2009-03-06 22:55:29.000000000 +0100
@@ -185,7 +185,7 @@
 target_alias = @target_alias@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
-SUBDIRS = libdc1394 examples
+SUBDIRS = libdc1394
 MAINTAINERCLEANFILES = Makefile.in aclocal.m4 configure config.h.in \
 	stamp-h.in
 
