--- src/raw1394.h.orig  2007-12-11 17:05:55.000000000 +0100
+++ src/raw1394.h       2007-12-11 17:06:50.000000000 +0100
@@ -40,7 +40,10 @@
 #define RAW1394_RCODE_TYPE_ERROR         0x6
 #define RAW1394_RCODE_ADDRESS_ERROR      0x7
 
-typedef u_int8_t  byte_t;
+/* So that the VLC build will not fail: */
+#ifndef _VLC_VLC_H
+  typedef u_int8_t  byte_t;
+#endif
 typedef u_int32_t quadlet_t;
 typedef u_int64_t octlet_t;
 typedef u_int64_t nodeaddr_t;
