--- orc-0.4.3/orc/orcprogram-c.c.orig	2009-12-12 01:28:26.000000000 +0100
+++ orc-0.4.3/orc/orcprogram-c.c	2010-03-05 23:12:44.000000000 +0100
@@ -65,7 +65,7 @@
   return 0;
 }
 
-const char *varnames[] = {
+static const char *varnames[] = {
   "d1", "d2", "d3", "d4",
   "s1", "s2", "s3", "s4",
   "s5", "s6", "s7", "s8",
--- orc-0.4.3/orc/orcprogram-c64x-c.c.orig	2009-11-20 16:46:56.000000000 +0100
+++ orc-0.4.3/orc/orcprogram-c64x-c.c	2010-03-05 23:13:01.000000000 +0100
@@ -104,7 +104,7 @@
   "t13", "t14", "t15", "t16"
 };
 
-void
+static void
 output_prototype (OrcCompiler *compiler)
 {
   OrcProgram *p = compiler->program;
