--- vlc-2.1.0/modules/text_renderer/freetype.c.orig	2013-08-18 23:06:27.145148641 +0200
+++ vlc-2.1.0/modules/text_renderer/freetype.c	2013-08-18 23:26:29.662181114 +0200
@@ -67,9 +67,9 @@
 # define SYSTEM_DEFAULT_MONOSPACE_FONT_FILE "/system/fonts/DroidSansMono.ttf"
 # define SYSTEM_DEFAULT_MONOSPACE_FAMILY "Droid Sans Mono"
 #else
-# define SYSTEM_DEFAULT_FONT_FILE "/usr/share/fonts/truetype/freefont/FreeSerifBold.ttf"
-# define SYSTEM_DEFAULT_FAMILY "Serif Bold"
-# define SYSTEM_DEFAULT_MONOSPACE_FONT_FILE "/usr/share/fonts/truetype/freefont/FreeMono.ttf"
+# define SYSTEM_DEFAULT_FONT_FILE "/usr/share/fonts/TTF/DejaVuSans-Bold.ttf"
+# define SYSTEM_DEFAULT_FAMILY "Sans Bold"
+# define SYSTEM_DEFAULT_MONOSPACE_FONT_FILE "/usr/share/fonts/TTF/DejaVuSansMono-Bold.ttf"
 # define SYSTEM_DEFAULT_MONOSPACE_FAMILY "Monospace"
 #endif
 
